flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```diff
@@ -396,6 +398,15 @@ public void visit(TemplateExpressionNode templateExpressionNode) {
         }
     }
 
+    @Override
+    public void visit(ByteArrayLiteralNode byteArrayLiteralNode) {
+        startNode(NodeKind.BINARY_DATA)
+                .metadata()
+                .description(BinaryData.DESCRIPTION)
```

Comment:
If the description does not depend on the node, you do not have to set it explicitly. 
```suggestion

```